acasclient/selfile.py

```diff
@@ -0,0 +1,851 @@
+    # is altered.
+    @property
+    def meta_has_changed(self):
+        """Return boolean,  True if the meta data changed after file parsing."""
+        return self._meta_has_changed
+    @meta_has_changed.setter
+    def meta_has_changed(self, value):
+        # Only set this if True so subsequent checks that are not changes
+        # can't unset.
+        if value:
+            self._meta_has_changed = value
+
+
+    @property
+    def experimental_meta_data(self):
+        """Experimental Meta Data (Meta Data)"""
+        return self._experimental_meta_data
+    @experimental_meta_data.setter
+    def experimental_meta_data(self, value):
+        self.meta_has_changed = (value != self._experimental_meta_data)
+        self._experimental_meta_data = value
+
+
+    @property
+    def format(self):
+        """Format (Meta Data)"""
+        return self._format
+    @format.setter
+    def format(self, value):
+        self.meta_has_changed = (value != self._format)
+        self._format = value
+
+
+    @property
+    def protocol_name(self):
+        """Protocol Name (Meta Data)"""
+        return self._protocol_name
+    @protocol_name.setter
+    def protocol_name(self, value):
+        self.meta_has_changed = (value != self._protocol_name)
+        self._protocol_name = value
+
+
+    @property
+    def experiment_name(self):
+        """Experiment Name (Meta Data)"""
+        return self._experiment_name
+    @experiment_name.setter
+    def experiment_name(self, value):
+        self.meta_has_changed = (value != self._experiment_name)
+        self._experiment_name = value
+    # Alias
+    name = experiment_name
+
+
+    @property
+    def scientist(self):
+        """Scientist (Meta Data)"""
+        return self._scientist
+    @scientist.setter
+    def scientist(self, value):
+        self.meta_has_changed = (value != self._scientist)
+        self._scientist = value
+
+
+    @property
+    def notebook(self):
+        """Notebook (Meta Data)"""
+        return self._notebook
+    @notebook.setter
+    def notebook(self, value):
+        self.meta_has_changed = (value != self._notebook)
+        self._notebook = value
+
+
+    @property
+    def page(self):
+        """Page (Meta Data)"""
+        return self._page
+    @page.setter
+    def page(self, value):
+        self.meta_has_changed = (value != self._page)
+        self._page = value
+
+
+    @property
+    def assay_date(self):
+        """Assay Date (Meta Data)"""
+        return self._assay_date
+    @assay_date.setter
+    def assay_date(self, value):
+        self.meta_has_changed = (value != self._assay_date)
+        self._assay_date = value
+
+
+    @property
+    def project(self):
+        """Project (Meta Data)"""
+        return self._project
+    @project.setter
+    def project(self, value):
+        self.meta_has_changed = (value != self._project)
+        self._project = value
+
+    @property
+    def id(self):
```

Comment:
I added id property as we have a use case where all classes used in an ETL script are required to include a id property

Reply:
'id' is reserved name in python

Reply:
Really? https://realpython.com/lessons/reserved-keywords/
It is not a python keyword. What do you mean by it being a reserved name?

Reply:
Apologies, I meant to call attention that you are shadowing a *keyword*.  
https://www.programiz.com/python-programming/keyword-list

Reply:
id is not listed on that page anywhere that I can find.

Reply:
Consensus here seems to be that it's ok: 
https://stackoverflow.com/questions/3497883/in-python-is-it-bad-to-create-an-attribute-called-id